### PR TITLE
6669 ppr api search 6728 6732

### PR DESCRIPTION
**Pull request by doug-lovett**

### CODE REVIEW

**Review by kialj876 — APPROVED** [review]

File: ppr-api/src/ppr_api/models/search_utils.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+# Account search history max result set size.
+ACCOUNT_SEARCH_HISTORY_MAX_SIZE = 500
+# Maximum number or results returned by search.
+SEARCH_RESULTS_MAX_SIZE = 1000
+
+# Result set size limit clause
+RESULTS_SIZE_LIMIT_CLAUSE = 'FETCH FIRST ' + str(SEARCH_RESULTS_MAX_SIZE) + ' ROWS ONLY'
+
+# Serial number search base where clause
+SERIAL_SEARCH_BASE = \
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/search_utils.py]: If you don't mind the extra whitespace, this works too
```
SERIAL_SEARCH_BASE = """
SELECT r.registration_type_cd,r.registration_ts AS base_registration_ts,
        sc.serial_type_cd,sc.serial_number,sc.year,sc.make,sc.model,
        r.registration_number AS base_registration_num,
        DECODE(serial_number, '?', 'EXACT', 'SIMILAR') AS match_type,
        fs.expire_date,fs.state_type_cd,sc.serial_id AS vehicle_id, sc.mhr_number  
FROM registration r, financing_statement fs, serial_collateral sc 
WHERE r.financing_id = fs.financing_id 
AND r.registration_type_cl IN ('PPSALIEN', 'MISCLIEN') 
AND r.base_reg_number IS NULL 
AND (fs.expire_date IS NULL OR fs.expire_date > ((SYSTIMESTAMP AT TIME ZONE 'UTC') - 30)) 
AND NOT EXISTS (SELECT r3.registration_id 
                    FROM registration r3 
                WHERE r3.financing_id = fs.financing_id 
                    AND r3.registration_type_cl = 'DISCHARGE' 
                    AND r3.registration_ts < ((SYSTIMESTAMP AT TIME ZONE 'UTC') - 30)) 
AND sc.financing_id = fs.financing_id 
AND sc.registration_id_end IS NULL
"""

File: ppr-api/tests/unit/api/test_search_results.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+from registry_schemas.example_data.ppr import SEARCH_SUMMARY
+
+from ppr_api.services.authz import STAFF_ROLE, COLIN_ROLE, PPR_ROLE
+from tests.unit.services.utils import create_header_account, create_header
+
+
+SAMPLE_JSON_SUMMARY = copy.deepcopy(SEARCH_SUMMARY)
+
+
+def test_search_detail_valid_200(session, client, jwt):
```
Comment (thorwolpert) [inline ppr-api/tests/unit/api/test_search_results.py:32]: you could collapse these with pytest parameterized tests

**Review by thorwolpert — COMMENTED** [review]
couple of considerations

File: ppr-api/src/ppr_api/models/search_utils.py (hunk 1 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/search_utils.py]: Query definitions updated as suggested.

File: ppr-api/tests/unit/api/test_search_results.py (hunk 2 above)
Comment (doug-lovett) [reply ppr-api/tests/unit/api/test_search_results.py:32]: Noted. I am refactoring as changes are made to unit test files. These tests already existed.

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**